infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultClusterManager.java

```diff
@@ -181,57 +229,161 @@ public TopologyConfigurationMsg queryTopologyConfig(boolean useCached) {
     }
 
     /**
-     * Enforce one of the standby Cluster's to become the new active cluster and current active to become standby
+     * Create a new topology config, which changes one of the standby as the active,
+     * and active as standby. Data should flow in the reverse direction.
      **/
-    public static TopologyDescriptor changeActiveCluster(TopologyConfigurationMsg topologyConfig) {
-        TopologyDescriptor topologyDescriptor = new TopologyDescriptor(topologyConfig);
-
-        // Convert the current active to standby
-        ClusterDescriptor oldActive = topologyDescriptor.getActiveClusters().values().iterator().next();
-        ClusterDescriptor newStandby = new ClusterDescriptor(oldActive, ClusterRole.STANDBY);
-
-        List<ClusterDescriptor> standbyClusters = Arrays.asList(newStandby);
-        ClusterDescriptor newPrimary = null;
-
-        for (ClusterDescriptor standbyCluster : topologyDescriptor.getStandbyClusters().values()) {
-            if (newPrimary == null) {
-                newPrimary = new ClusterDescriptor(standbyCluster, ClusterRole.ACTIVE);
+    public TopologyDescriptor generateConfigWithRoleSwitch() {
+        TopologyDescriptor currentConfig = new TopologyDescriptor(topologyConfig);
+
+        List<ClusterDescriptor> newActiveClusters = new ArrayList<>();
+        List<ClusterDescriptor> newStandbyClusters = new ArrayList<>();
+        currentConfig.getActiveClusters().values().forEach(activeCluster ->
+                newStandbyClusters.add(new ClusterDescriptor(activeCluster, ClusterRole.STANDBY)));
+        for (ClusterDescriptor standbyCluster : currentConfig.getStandbyClusters().values()) {
+            if (newActiveClusters.isEmpty()) {
+                newActiveClusters.add(new ClusterDescriptor(standbyCluster, ClusterRole.ACTIVE));
             } else {
-                standbyClusters.add(standbyCluster);
+                newStandbyClusters.add(new ClusterDescriptor(standbyCluster, ClusterRole.STANDBY));
             }
         }
 
-        TopologyDescriptor newSiteConf = new TopologyDescriptor(1L, Arrays.asList(newPrimary), standbyClusters);
-        return newSiteConf;
+        return new TopologyDescriptor(++configId, newActiveClusters, newStandbyClusters);
+    }
+
+    /**
+     * Create a new topology config, which marks all standby cluster as active on purpose.
+     * System should drop messages between any two active clusters.
+     **/
+    public TopologyDescriptor generateConfigWithAllActive() {
```

Comment:
can you add an annotation or comment that this in only for testing?

Reply:
The same. I think I will refactor this part in the next PR.